Assets/MixedRealityToolkit.Tests/PlayModeTests/InteractableTests.cs

```diff
@@ -61,6 +61,146 @@ public IEnumerator TestAddInteractableAtRuntime()
             yield return null;
         }
 
+        /// <summary>
+        /// Tests that an interactable component can be added to a GameObject
+        /// at runtime, and an OnClick event handler can be added
+        /// </summary>
+        /// <returns></returns>
+        [UnityTest]
+        public IEnumerator TestAddOnClickRuntime()
+        {
+            var cube = GameObject.CreatePrimitive(PrimitiveType.Cube);
+            cube.transform.position = Vector3.forward;
+            PlayModeTestUtilities.PushHandSimulationProfile();
+            PlayModeTestUtilities.SetHandSimulationMode(HandSimulationMode.Gestures);
+
+            // This should not throw an exception
+            var interactable = cube.AddComponent<Interactable>();
+
+            // Subscribe to interactable's on click so we know the click went through
+            bool wasClicked = false;
+            interactable.OnClick.AddListener(() => { wasClicked = true; });
+
+            TestHand th = new TestHand(Handedness.Right);
+            yield return th.Show(Vector3.zero);
+
+            CameraCache.Main.transform.LookAt(interactable.transform);
+            yield return th.SetGesture(ArticulatedHandPose.GestureId.Pinch);
+            yield return th.SetGesture(ArticulatedHandPose.GestureId.Open);
+
+            Assert.True(wasClicked);
+
+            // clean up
+            GameObject.Destroy(cube);
+            PlayModeTestUtilities.PopHandSimulationProfile();
+
+            yield return null;
+        }
+
+        /// <summary>
+        /// Tests that an interactable component can be added to a GameObject
+        /// at runtime, and all receivers that extend ReceiverBase
+        /// and have event handlers can easily be added, and work
+        /// </summary>
+        /// <returns></returns>
+        [UnityTest]
+        public IEnumerator TestRuntimeEvents()
+        {
+            var cube = GameObject.CreatePrimitive(PrimitiveType.Cube);
+            cube.transform.position = Vector3.right;
+
+            var interactable = cube.AddComponent<Interactable>();
+
+            // OnFocus
+            var onFocusReceiver = interactable.AddReceiver<InteractableOnFocusReceiver>();
+
+            bool didHover = false;
+            bool didUnHover = false;
+            onFocusReceiver.OnFocusOn.AddListener(() => didHover = true);
+            onFocusReceiver.OnFocusOff.AddListener(() => didUnHover = true);
+            CameraCache.Main.transform.LookAt(interactable.transform);
+            yield return null;
+            CameraCache.Main.transform.LookAt(Vector3.forward);
+            yield return null;
+            Assert.True(didHover, "Interactable did not receive hover event");
+            Assert.True(didUnHover, "Interactable did not receive un-hover event");
```

Comment:
If we already have coverage of this in another area feel free to ignore this comment - note that it might be better to have another line like:

Assert.True(didHover, "Interactable did not receive hover event");
Assert.False(didUnHover, "Interactable shouldn't have received unhover event");

After looking at the interactable, to show that the right operations are occuring in the right way.

Reply:
done. good idea.